File: Lab04 D SDI/Parameters.pas
unit Parameters;

interface

uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes, Vcl.Graphics,
  Vcl.Controls, Vcl.Forms, Vcl.Dialogs, Vcl.StdCtrls, Vcl.ExtCtrls;

type
  TForm_params = class(TForm)
    CB1: TCheckBox;
    Panel: TPanel;
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form_params: TForm_params;

implementation
Uses Main;
{$R *.dfm}

procedure TForm_params.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  FormMain.Parameters.enabled:=true;
  action:=cafree;
end;

end.

File: Lab03 CB MDI/Parameters.cpp
// ---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Parameters.h"
#include "Main.h"
// ---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm_params *Form_params;
extern cb;

// ---------------------------------------------------------------------------
__fastcall TForm_params::TForm_params(TComponent* Owner) : TForm(Owner) {
}
// ---------------------------------------------------------------------------

void __fastcall TForm_params::CB1Click(TObject *Sender) {
	cb = CB1->Checked;
}
// ---------------------------------------------------------------------------

void __fastcall TForm_params::Form_params_Create(TObject *Sender) {
	CB1->Checked = cb;
}
// ---------------------------------------------------------------------------

void __fastcall TForm_params::Form_params_Close(TObject *Sender,
	TCloseAction &Action) {
	Form_main->Parameters->Enabled = true;
	Action = caFree;
}
// ---------------------------------------------------------------------------
